src/main/java/io/gingersnapproject/cdc/ManagedEngine.java

```diff
@@ -38,25 +32,17 @@ public class ManagedEngine {
    @Inject
    Configuration config;
 
-   @Inject @All List<CacheService> services;
-   @Inject NotificationManager eventing;
+   @Inject
+   CacheService cacheService;
 
-    CacheService findCacheService(URI uri) {
-      for (CacheService cacheService : services) {
-         if (cacheService.supportsURI(uri)) {
-            return cacheService;
-         }
-      }
-      throw new IllegalArgumentException("Unsupported URI received: " + uri + " ensure service is running if correct!");
-   }
+   @Inject NotificationManager eventing;
 
    public void start(@Observes StartupEvent ignore) {
       log.info("Starting service");
+      cacheService.start();
```

Comment:
Is there a better way we can handle this? It doesn't seem great controlling a bean's lifecycle from another bean.

Reply:
You can add the same events to the CacheService impls instead. Then they can control their own lifecycle. Then iiuc  quarkus will order based on dependency. So in this case the CacheService is a dependency of the Engine so they should be stopped in appropriate order.

Reply:
I did try that originally, but on some startups the CacheService wasn't initated before `ManagedEngine#startStopEngine` was called.

Reply:
Weird.. that sounds like a bug in Quarkus. I would expect dependencies to have start method called in dependent order. But maybe they don't :(

Reply:
Another alternative would be creating a method with the `@Produces`, starting the service, and returning it.

Reply:
I have got it working now, I just needed to make the HotRodService `@ApplicationScoped` and pass it as a parameter of the ManagedEngine#start